### PR TITLE
refactor bench tests in algebra

### PR DESCRIPTION
**Pull request by AndreyMlashkin**

### CODE REVIEW

**Review by x-mass — APPROVED** [review]

File: test/bench_test/CMakeLists.txt [hunk 1]
```diff
@@ -12,11 +12,11 @@ if(NOT Boost_UNIT_TEST_FRAMEWORK_FOUND)
...
 endif()
 
-add_custom_target(algebra_runtime_bench_tests)
+add_custom_target(${CURRENT_PROJECT_NAME}_bench)
 
-macro(define_runtime_algebra_test name)
-    set(test_name "algebra_${name}_bench_test")
-    add_dependencies(algebra_runtime_bench_tests ${test_name})
+macro(define_bench name)
+    set(test_name "algebra_${name}_bench")
```
Comment (x-mass) [inline test/bench_test/CMakeLists.txt:24]: As for me, it's better to name it with `_test` suffix, but never mind

File: bench_test/CMakeLists.txt [hunk 2]
```diff
@@ -12,11 +12,17 @@ if(NOT Boost_UNIT_TEST_FRAMEWORK_FOUND)
...
-    add_dependencies(algebra_runtime_bench_tests ${test_name})
+                       ${CMAKE_WORKSPACE_NAME}::multiprecision
+
+                       ${Boost_LIBRARIES})
+
+add_custom_target(${CURRENT_PROJECT_NAME}_bench)
+
+macro(define_bench name)
+    set(test_name "algebra_${name}_bench")
+    add_dependencies(${CURRENT_PROJECT_NAME}_bench ${test_name})
```
Comment (x-mass) [inline bench_test/CMakeLists.txt:25]: I think it's better to have `_test` suffix, but never mind

File: bench_test/CMakeLists.txt [hunk 3]
```diff
@@ -31,7 +37,7 @@ macro(define_runtime_algebra_test name)
     cm_test(NAME ${test_name} SOURCES ${name}.cpp ARGS ${additional_args})
 
     target_include_directories(${test_name} PRIVATE
-                               "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+                               "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>"
```
Comment (x-mass) [inline bench_test/CMakeLists.txt:40]: It's a bed pattert to use any files not from the current dir or its subdirs in CMake. Move those includes into a different interface target

**Review by x-mass — CHANGES_REQUESTED** [review]

File: test/bench_test/CMakeLists.txt (hunk 1 above)
Comment (AndreyMlashkin) [reply test/bench_test/CMakeLists.txt:24]: Those tests shall not be executed in the normal workflow

File: bench_test/CMakeLists.txt (hunk 3 above)
Comment (AndreyMlashkin) [reply bench_test/CMakeLists.txt]: fixed in 087efb2c2cdb89e3525bd999b7ceba2e9f9a85ce

File: CMakeLists.txt [hunk 4]
```diff
@@ -87,3 +87,7 @@ endif()
 if(BUILD_EXAMPLES)
     add_subdirectory(example)
 endif()
+
+if(BUILD_BENCH_TESTS)
+    add_subdirectory(bench_test)
+endif()
```
Comment (x-mass) [inline CMakeLists.txt:93]: Check this issue
https://github.com/NilFoundation/crypto3/issues/146

I've already made it in hashes
https://github.com/NilFoundation/crypto3-hash/blob/cc48e2ead139c7c7f190b3945c14c2237948805c/CMakeLists.txt#L282

**Review by x-mass — CHANGES_REQUESTED** [review]

File: bench_test/CMakeLists.txt [hunk 5]
```diff
@@ -12,11 +12,17 @@ if(NOT Boost_UNIT_TEST_FRAMEWORK_FOUND)
...
 
-add_custom_target(algebra_runtime_bench_tests)
+cm_test_link_libraries(${CMAKE_WORKSPACE_NAME}_${CURRENT_PROJECT_NAME}
 
-macro(define_runtime_algebra_test name)
-    set(test_name "algebra_${name}_bench_test")
-    add_dependencies(algebra_runtime_bench_tests ${test_name})
+                       ${CMAKE_WORKSPACE_NAME}::multiprecision
+
+                       ${Boost_LIBRARIES})
```
Comment (x-mass) [inline bench_test/CMakeLists.txt:19]: These are propageted through `${CMAKE_WORKSPACE_NAME}_${CURRENT_PROJECT_NAME}`

File: bench_test/CMakeLists.txt [hunk 6]
```diff
@@ -31,9 +37,6 @@ macro(define_runtime_algebra_test name)
     cm_test(NAME ${test_name} SOURCES ${name}.cpp ARGS ${additional_args})
 
     target_include_directories(${test_name} PRIVATE
-                               "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-                               "$<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>"
-
                                ${Boost_INCLUDE_DIRS})
```
Comment (x-mass) [inline bench_test/CMakeLists.txt:40]: No need in this inlcudes. They are already declared for tested target:
https://github.com/NilFoundation/crypto3-algebra/blob/087efb2c2cdb89e3525bd999b7ceba2e9f9a85ce/CMakeLists.txt#L67-L71

**Review by x-mass — CHANGES_REQUESTED** [review]

**Pull request closed without merging**